c++/include/orc/Vector.hh

```diff
@@ -88,6 +89,11 @@ namespace orc {
      */
     virtual bool hasVariableLength();
 
+    /**
+     * Decode dictionary into vector batch.
+     */
+    virtual void decodeDictionary() {}
```

Comment:
We need to implement this for struct, map, list and union type by calling it recursively.

Reply:
BTW, please add a test case to make sure it works as expected.